### PR TITLE
Allow diffing tags

### PR DESCRIPTION
**Pull request by johnoliver**

### CODE REVIEW

File: git-hg/diff.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@
 #
 ################################################################################
 
-set -euo
+set -euxo
```
Comment (karianna) [inline git-hg/diff.sh]: Shouldn't `o` have an option name after it?

File: git-hg/diff.sh [hunk 2]
```diff
@@ -53,6 +54,17 @@ function cloneRepos() {
...
+
+  if [ -n "${tag}" ]; then
+    cd openjdk-hg
+    hg update $tag
+    cd ../openjdk-git
+    git fetch --all
+    git checkout $tag
+    cd ..
+  fi
+
```
Comment (karianna) [inline git-hg/diff.sh]: Kill extra whitespace

File: git-hg/diff.sh [hunk 3]
```diff
@@ -53,6 +54,17 @@ function cloneRepos() {
 
   git clone -b master "https://github.com/AdoptOpenJDK/openjdk-${git_repo_version}.git" openjdk-git || exit 1
   hg clone "https://hg.openjdk.java.net/${hg_root_forest}/${hg_repo_version}" openjdk-hg || exit 1
+
+  if [ -n "${tag}" ]; then
+    cd openjdk-hg
+    hg update $tag
+    cd ../openjdk-git
+    git fetch --all
+    git checkout $tag
```
Comment (karianna) [inline git-hg/diff.sh]: We're happy that we've left this function with git pointing at the tag (do we need to reset later?)

File: git-hg/diff.sh [hunk 4]
```diff
@@ -65,7 +77,10 @@ function updateMercurialClone() {
 }
 
 function runDiff() {
-  diffNum=$(diff -rq openjdk-git openjdk-hg -x '.git' -x '.hg' -x '.hgtags' -x '.hgignore' -x 'get_source.sh' -x 'README.md' | wc -l)
+
+  diff -rq openjdk-git openjdk-hg -x '.git' -x '.hg' -x '.hgtags' -x '.hgignore' -x 'get_source.sh' -x 'README.md' > changes.diff
+
+  diffNum=$(wc -l < changes.diff)
 
   if [ "$diffNum" -gt 0 ]; then
```
Comment (karianna) [inline git-hg/diff.sh]: Do we want to log somewhere that the `changes.diff` file can be inspected?

Do we need to delete that file later?

File: git-hg/diff.sh (hunk 3 above)
Comment (johnoliver) [reply git-hg/diff.sh]: the workspace gets wiped out every time

File: git-hg/diff.sh (hunk 4 above)
Comment (johnoliver) [reply git-hg/diff.sh]: can archive it in the job config

File: git-hg/diff.sh [hunk 5]
```diff
@@ -42,6 +42,8 @@ checkArgs $@
 git_repo_version=$1
 hg_root_forest=${2:-${1}}                  # for backwards compatibility
 hg_repo_version=${3:-${hg_root_forest}}    # for backwards compatibility
+tag=${4:-} # tag to check
+expectedDiffLimit=${5:-0} # expected diff limit
```
[the remark this answers is not in the document]
Comment (karianna) [reply git-hg/diff.sh]: Ah, number of diffs that we allow before we say "there's a problem" - perhaps update comment with that

File: git-hg/diff.sh [hunk 6]
```diff
@@ -53,53 +55,56 @@ function cloneRepos() {
...
+    cd openjdk-hg
+    hg update $tag
+    cd ../openjdk-git
+    git fetch --all
+    git checkout $tag
+    cd ..
+  fi
 }
 
 function updateMercurialClone() {
```
[the remark this answers is not in the document]
Comment (johnoliver) [reply git-hg/diff.sh]: should do yes

**Pull request merged**